### ISSUE TITLE
Add many DataPlotlyDock to build a dashboard with dataPlotly

### ISSUE THREAD

**Issue by jdlom** [ghtmtt/DataPlotly#313]
Hi,
I try to implement some new feature in order to build a dashboard with this amazing plugin.

I want to do something like that :

<img width="1920" alt="qgis-ltr-bin_WEcd2M46cj" src="https://user-images.githubusercontent.com/8641271/224079322-fe662223-f925-4c5c-b860-7ecd3dbc2b1b.png">

I have a function to add a new dock_widget : 
```python
    def addNewDock(self, title='DataPlotly', key='main_dock', hide = True, message_bar = None):
        if key in self.dock_widgets:
            self.iface.messageBar().pushWarning(self.tr('Warning'), self.tr(f'DataPlotlyDock can not be added because {key} is already present'))
            return
        message_bar = message_bar or self.iface.messageBar()
        dock = DataPlotlyDock(title = title, message_bar=message_bar)
        self.dock_widgets[key] = dock
        self.iface.addDockWidget(Qt.RightDockWidgetArea, dock)
        if hide:
            dock.hide()
        return dock
```

Unfortunely, whereas every `DataPlotly.gui.plot_settings_widget.DataPlotlyPanelWidget` in `DataPlotly.gui.dock.DataPlotlyDock` have its own settings, when plot are updated, all plots are updated with the last created plot .

Do you think this could be implemented without effort ?

**Comment by ghtmtt** [comment]
that would be an amazing enhancement! Well I wouldn't say without any efforts, but definitely doable. 

I think be main problem is that the buttons among each dock will call the settings of the last plot chosen. Let me also ping @nyalldawson.. Do you have some ideas (or existing API) that we can have a look at?

**Comment by jdlom** [comment]
Cool. I would be happy to contribute even if my skills are limited.
Anyways, I have tested to add a `dock_id` argument to settings to save and restore from project and...
 it works like a charm
![qgis-ltr-bin_l8CzGsM65G](https://user-images.githubusercontent.com/8641271/224110901-ae88630b-5697-495a-ac80-b6c2a20475cf.gif)

**Comment by ghtmtt** [comment]
That's great! Could you please try:

* single docks: be sure that dock related settings are taken into account only  for that dock
* can you make some test also in the print layout composer? Code called there is different, but some tests will be more than welcome
* we have also to try on project saved with old versions of the plugin: if old qgs project files are ignoring the `dock_id` and are able to restore the single dock than perfect, if they are not able to understand the `dock_id` than a stack trace can be raised

**Comment by jdlom** [comment]
In the gif above, docks are already present. It wase my first tests. Since, I assumed that the docks except the main dock should be closed automatically when the project is closed. I would like to have your point of view. 

All settings are saved in qgis project like that `<DataPlotly_dockid>`
> That's great! Could you please try:
> 
>     * single docks: be sure that dock related settings are taken into account only  for that dock

It works for a single dock.

>     * can you make some test also in the print layout composer? Code called there is different, but some tests will be more than welcome

I will test ! But it's seems complicated (for me) to add more docks in the print layout composer. I didn't find a proper way to clean the toolbar in the print layout. But it will be nice to have more 
> 
>     * we have also to try on project saved with old versions of the plugin: if old qgs project files are ignoring the `dock_id` and are able to restore the single dock than perfect, if they are not able to understand the `dock_id` than a stack trace can be raised

It should be done with minor adjustments.

I need also to store the setting's dock ( in particular the dock position in the main window). For the moment it store only settings plot.

Before starting a new PR, I will wait for @nyalldawson response.

**Comment by jdlom** [comment]
Ok. I just figure out why the plots are the same when  the plot view is refrehed  !  All plots share the same `output_file`. 
https://github.com/ghtmtt/DataPlotly/blob/f4cc7d4165a4e69de6abae78a6d53ad5489f80f8/DataPlotly/core/plot_factory.py#L679

**Comment by jdlom** [comment]
![qgis-ltr-bin_60dtNohrnw](https://user-images.githubusercontent.com/8641271/224515701-090812f4-3aca-4379-85a0-9fecbfaef513.gif)
Now it's working. I'm preparing a PR for you soon.

**Comment by ghtmtt** [comment]
AWESOME!

Quick question: in the upper gif you are testing the selection on a layer and the corresponding update of the plots. If the plots refer to different layers, right now the selection on plot items highlights the corresponding feature on the map. Is this behavior the same with many docks?

**Comment by jdlom** [comment]
I think it's working.I will test in depth to be sure of what I say
![qgis-ltr-bin_x35ALEbghU](https://user-images.githubusercontent.com/8641271/224679603-c9ea5624-3c67-44d8-96e2-ba2729c03dba.gif)

**Comment by jdlom** [comment]
I would like to have your opinion. How do you want the user manages the creation of docks ? I though adding a new entry in project settings with a tablewidget to add or remove dock.

<img width="592" alt="qgis-ltr-bin_B0RU6gYCew" src="https://user-images.githubusercontent.com/8641271/224711289-d3b20669-4284-4b75-a073-6f81f14d38e8.png">

**Comment by ghtmtt** [comment]
> I would like to have your opinion. How do you want the user manages the creation of docks ? I though adding a new entry in project settings with a tablewidget to add or remove dock.

So, to create a new dock the user has to go to the project settings? What if we add another button to the main one, to add another dock? Then managing all the docks within the project properties would be great and maybe easier for the users. What do you think?

**Comment by ghtmtt** [comment]
> You want the button to directly add a dock or access the settings window ?

I'd say adding a dock. Then we can add another button (if needed) or specify in the documentation, that there are also the settings of the plugin in the project settings dialog.

**Comment by ghtmtt** [comment]
@jdlom do you need some help to make the PR? Don't hesitate!
